packages/rpc-types/src/lamports.ts

```diff
@@ -46,18 +48,34 @@ export function lamports(putativeLamports: bigint): LamportsUnsafeBeyond2Pow53Mi
     return putativeLamports;
 }
 
-export function getLamportsEncoder(): FixedSizeEncoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
+export function getLamportsEncoder(
+    innerEncoder?: NumberEncoder,
+): FixedSizeEncoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
+    if (innerEncoder) {
+        return fixEncoderSize(innerEncoder, 8);
+    }
     return getMemoizedU64Encoder();
 }
 
-export function getLamportsDecoder(): FixedSizeDecoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
+export function getLamportsDecoder(
+    innerDecoder?: NumberDecoder,
+): FixedSizeDecoder<LamportsUnsafeBeyond2Pow53Minus1, 8> {
+    if (innerDecoder) {
```

Comment:
It might be best to offer two helper methods. One that accepts an explicit number codec to wrap and one that uses the memoized u64 codec as the number codec (which can delegate to the first helper function). That way the latter is flexible and is as "fixed" or "variable" as the provided number codec. Whereas the second one is a fixed codec of 8 bytes, always (and this can be used in most places within this library).

Reply:
Is that to preserve the `FixedSizeDecoder<LamportsUnsafeBeyond2Pow53Minus1, 8>` output type when this is called without an inner de/encoder?